### PR TITLE
Add product card styling.

### PR DESCRIPTION
**Pull request by madeincosmos**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:
- Update product cards on the WooCommerce > Extensions page in WP Admin to match current designs.

### How to test the changes in this Pull Request:

1. Check out this branch on local environment.
2. Navigate to one of marketplace category pages, i.e. `wp-admin/admin.php?page=wc-addons&section=payment-gateways`  (nav menu isn't implemented at the moment
3. Check product card and make sure it looks good on all screen sizes

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable? - e2e tests are currently failing because the WooCommerce.com Subscriptions screen is now in a different place, these changes are addressed in #30380.
* [x] Have you successfully run tests with your changes locally?

### FOR PR REVIEWER ONLY:

* [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Review by andfinally — COMMENTED** [review]
Looking great, thanks! There are a few things I still need to check. I think we probably need to retain the max-width 1200px on the list of products and align it left, like in the current page. So we may want to adjust the width of the cards a little bit to allow for that greater width.

File: assets/css/admin.scss [hunk 1]
```diff
@@ -537,72 +537,85 @@
...
 			min-width: 280px;
 			min-height: 220px;
-			flex: 1;
 			overflow: hidden;
-			background: #f5f5f5;
-			box-shadow: inset 0 1px 0 rgba(255, 255, 255, 0.2),
-				inset 0 -1px 0 rgba(0, 0, 0, 0.1);
+			padding: 0;
+			vertical-align: top;
+			width: 45%;
```
[the remark this answers is not in the document]
Comment (madeincosmos) [reply assets/css/admin.scss:574]: Added max width on product card and an extra wrapper to make the product list look good on all screen sizes:

**Comment by tgglv** [comment]
Hey @madeincosmos, just started to see this PR and noticed that E2E tests [failed](https://github.com/woocommerce/woocommerce/pull/30410/checks?check_run_id=3278191858):
```
a.nav-tab (text: "WooCommerce.com Subscriptions") not found
```
It looks like the expected element is missed. It might happen coincidentally. Can you take a look?

**Comment by madeincosmos** [comment]
> Hey @madeincosmos, just started to see this PR and noticed that E2E tests [failed](https://github.com/woocommerce/woocommerce/pull/30410/checks?check_run_id=3278191858):

Hi @tgglv, the e2e tests are failing because we changed the marketplace page structure. I've updated the tests in a different branch that is currently merged into ` update/marketplace-1`, I can try to rebase this one if that's needed.

**Review by andfinally — APPROVED** [review]
Looks lovely! My apologies Maria, I pushed some additions to this PR before I saw your [comment](https://github.com/woocommerce/woocommerce/pull/30410#discussion_r683270518), hope this is consistent with your work. I made some tweaks to the page layout.

- Made the wrapper full width, so the purple header stretches across the screen.
- Aligned the products to the left.
- Used `calc` for product card width, so any trailing odd-numbered card keeps the same width as the others.
- Addressed some stylelint warnings in `admin.scss`.

The PHP7.4 WP Nightly test is repeatedly failing, something to do with composer dependencies – I don't believe it's related to the changes in this PR.

**Comment by madeincosmos** [comment]
Thanks @andfinally, I like your approach better than the previous calculations based on margins. I can see in the designs that there's 84px space between WordPress menu and product list, so I've added this on larger screens. Other than this, I think we're good to.

**Pull request merged**